src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -1459,6 +1459,41 @@ .locals init (int V_0)
 }");
         }
 
+        [Fact]
+        public void InParamEnum()
+        {
+            var text = @"
+class Program
+{
+    static void Main()
+    {
+        Enum1 v = Enum1.Value1;
+        System.Console.WriteLine(M(v));
+    }
+
+    static string M(in Enum1 x) => x.ToString();
```

Comment:
Should add unit tests for the other cases in the issue, specifically something like:
```csharp
static string M1(in Enum1 x) => x.ToString();
static string M2<T>(in T x) where T : Enum => x.ToString();
static string M3<T>(in T x) where T : struct, Enum => x.ToString();
```

Reply:
Done, had to amend the logic to make the type params work.